kuma/api/v1/tests/test_search.py

```diff
@@ -57,45 +57,68 @@ def search(self, *args, **kwargs):
         return result
 
 
-def test_search_basic_match(user_client, settings):
+@pytest.fixture
+def mock_elasticsearch():
     fake_elasticsearch = FindEverythingFakeElasticsearch()
+    with patch("elasticsearch_dsl.search.get_connection") as get_connection:
+        get_connection.return_value = fake_elasticsearch
+        yield fake_elasticsearch
```

Comment:
So I noticed, the way I was doing it before was busted. 
When I added the now 2nd unit test with mocks, the second test would get the results from the fixture mocking in the first test. 
There are two unit tests (now): `test_search_basic_match` and `test_search_case_insensitive_locale` and because of connection pooling, what happens is that in the second test the mocking is already done for you and it reuses the first mocking. 
So I discovered a new, better way, to mock the connection. This feels much better.